sharding/proposer/service.go

```diff
@@ -1,29 +1,27 @@
+// Package notary defines all relevant functionality for a Proposer actor
+// within the minimal sharding protocol.
 package proposer
 
 import (
 	"github.com/ethereum/go-ethereum/log"
-	"github.com/ethereum/go-ethereum/sharding/database"
-	"github.com/ethereum/go-ethereum/sharding/node"
+	"github.com/ethereum/go-ethereum/sharding"
 )
 
 // Proposer holds functionality required to run a collation proposer
 // in a sharded system. Must satisfy the Service interface defined in
 // sharding/service.go.
 type Proposer struct {
-	node    node.Node
-	shardDB database.ShardBackend
+	client   sharding.SMCClient
+	shardp2p sharding.ShardP2P
+	txpool   sharding.TXPool
 }
 
 // NewProposer creates a struct instance. It is initialized and
 // registered as a service upon start of a sharding node.
 // Has access to the public methods of this node.
```

Comment:
Not sure what `Has access to the public methods of this node` means. Can we make it more clear?